lib/rubocop/cop/lint/useless_trailing_comma.rb

```diff
@@ -0,0 +1,113 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # This cop checks whether trailing commas in block arguments are
+      # required. Blocks with only one argument and a trailing comma require
+      # that comma to be present. Blocks with more than one argument never
+      # require a trailing comma.
+      #
+      # @example
+      #   # bad
+      #   add { |foo, bar,| foo + bar }
+      #
+      #    # good
+      #   add { |foo, bar| foo + bar }
+      #
+      #   # good
+      #   add { |foo,| foo }
+      #
+      #   # good
+      #   add { foo }
+      #
+      #   # bad
+      #  add do |foo, bar,|
+      #    foo + bar
+      #   end
+      #
+      #  # good
+      #  add do |foo, bar|
+      #    foo + bar
+      #  end
+      #
+      #  # good
+      #   add do |foo,|
+      #    foo
+      #  end
+      #
+      #  # good
+      #  add do
+      #     foo + bar
+      #  end
+      class UselessTrailingComma < Cop
+        include RangeHelp
+
+        MSG = 'Block arguments do not require a trailing comma except when ' \
+              'only a single argument is present.'
+
+        ARG_MSG = 'Useless trailing comma present in block arguments.'
+
+        def on_block(node)
+          return unless useless_trailing_comma?(node)
+
+          add_offense_for_args(node, node.arguments, ARG_MSG)
+        end
+
+        def autocorrect(node)
+          lambda do |corrector|
+            return unless useless_trailing_comma?(node)
+
+            end_pos = range_with_surrounding_space(
+              range: node.arguments.source_range,
+              side: :right,
+              newlines: false
+            ).end_pos
+            range = range_between(node.loc.begin.end.begin_pos, end_pos)
+            corrector.replace(range, block_arg_string(node, node.arguments))
+          end
+        end
+
+        private
+
+        def block_arg_string(node, args)
```

Comment:
This seems to destroy semicolons that separate block-local variables:
```
7637.rb:3:7: W: [Corrected] Lint/UselessTrailingComma: Useless trailing comma present in block arguments.
add { |foo, bar,; baz| foo + bar }
      ^^^^^^^^^^^^^^^^
====================
add { |foo, bar, baz| foo + bar }
```
I also don't like the fact that this cop changes the whitespace between the arguments, i.e. might destroy an intentional formatting (alignment, linebreaks, etc.). In my opinion, this cop should only remove a single comma and do nothing more.

Reply:
Reworked to touch only a single comma.